Cannot run `cargo test -p vortex-python`

### What happened?

If I run `cargo test -p vortex-python` on its own on develop I get weird errors:

```
vortex on  develop [$] is 📦 v0.1.0 via  v25.9.0 via 🐍 v3.14.4 via 🦀 v1.91.0 on ☁️  (us-east-1) [X] took 1m22s
❯ cargo t -p vortex-python
   Compiling vortex-python v0.1.0 (/Users/connor/spiral/vortex-data/vortex/vortex-python)
error[E0432]: unresolved import `vortex::array::assert_arrays_eq`
  --> vortex-python/src/arrays/range_to_sequence.rs:76:9
   |
76 |     use vortex::array::assert_arrays_eq;
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `assert_arrays_eq` in `array`

warning: unused import: `vortex::array::IntoArray as _`
  --> vortex-python/src/arrays/range_to_sequence.rs:75:9
   |
75 |     use vortex::array::IntoArray as _;
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(unused_imports)]` (part of `#[warn(unused)]`) on by default

warning: unused import: `vortex::buffer::buffer`
  --> vortex-python/src/arrays/range_to_sequence.rs:77:9
   |
77 |     use vortex::buffer::buffer;
   |         ^^^^^^^^^^^^^^^^^^^^^^
```

### Steps to reproduce

See above

### Environment

MacOS

### Additional context

_No response_

--- a/vortex-python/Cargo.toml
+++ b/vortex-python/Cargo.toml
@@ -52,3 +52,6 @@ tokio = { workspace = true, features = ["fs", "rt-multi-thread"] }
 url = { workspace = true }
 vortex = { workspace = true, features = ["object_store", "tokio"] }
 vortex-tui = { workspace = true, optional = true }
+
+[dev-dependencies]
+vortex-array = { path = "../vortex-array", features = ["_test-harness"] }

--- a/vortex-python/src/arrays/range_to_sequence.rs
+++ b/vortex-python/src/arrays/range_to_sequence.rs
@@ -57,7 +57,7 @@ fn range_len(start: isize, stop: isize, step: isize) -> Option<usize> {
             usize::try_from(len).vortex_expect("stop >= start, step > 0, so len is non-negative");
         Some(len)
     } else {
-        assert!(step != 0);
+        assert_ne!(step, 0);
 
         if stop > start {
             return None;

